src/dialog/dlgreplacecuecolor.cpp

```diff
@@ -0,0 +1,382 @@
+#include <QDialogButtonBox>
+#include <QMessageBox>
+#include <QtConcurrent>
+
+#include "engine/controls/cuecontrol.h"
+#include "library/dao/cuedao.h"
+#include "library/queryutil.h"
+#include "preferences/colorpalettesettings.h"
+#include "util/color/predefinedcolorpalettes.h"
+
+namespace {
+
+constexpr int kColorButtonLightnessThreshold = 0x80;
+const QString kColorButtonStyleSheetLight = QStringLiteral(
+        "QPushButton { background-color: %1; }");
+const QString kColorButtonStyleSheetDark = QStringLiteral(
+        "QPushButton { background-color: %1; color: white; }");
+
+void setButtonColor(QPushButton* button, const QColor& color) {
+    button->setText(color.name());
+    button->setStyleSheet(
+            (Color::isDimmColor(color)
+                            ? kColorButtonStyleSheetDark
+                            : kColorButtonStyleSheetLight)
+                    .arg(color.name()));
+}
+
+typedef struct {
+    int id;
+    TrackId trackId;
+    mixxx::RgbColor color;
+} CueDatabaseRow;
+
+} // namespace
+
+DlgReplaceCueColor::DlgReplaceCueColor(
+        UserSettingsPointer pConfig,
+        mixxx::DbConnectionPoolPtr dbConnectionPool,
+        TrackCollectionManager* pTrackCollectionManager,
+        QWidget* pParent)
+        : QDialog(pParent),
+          m_pConfig(pConfig),
+          m_pDbConnectionPool(dbConnectionPool),
+          m_pTrackCollectionManager(pTrackCollectionManager),
+          m_bDatabaseChangeInProgress(false),
+          m_bDatabaseChanged(false),
+          m_pNewColorMenu(new QMenu(this)),
+          m_pCurrentColorMenu(new QMenu(this)),
+          m_pStyle(QStyleFactory::create(QStringLiteral("fusion"))) {
+    setupUi(this);
+    setWindowModality(Qt::ApplicationModal);
+
+    spinBoxHotcueIndex->setMaximum(NUM_HOT_CUES);
+
+    // Unfortunately, not all styles supported by Qt support setting a
+    // background color for QPushButtons (see
+    // https://bugreports.qt.io/browse/QTBUG-11089). For example, when using
+    // the gtk2 style all color buttons would be just grey. It's possible to
+    // work around this by modifying the button border with a QSS stylesheet,
+    // so that the QStyle will be overwritten, but as a sane default for skins
+    // without styles for WColorPicker, we're setting the platform-independent
+    // "Fusion" style here. This will make the buttons look slightly different
+    // from the rest of the application (when not styled via QSS), but that's
+    // better than having buttons without any colors (which would make the
+    // color picker unusable).
+    pushButtonNewColor->setStyle(m_pStyle);
+    pushButtonCurrentColor->setStyle(m_pStyle);
+
+    // Set up new color button
+    ColorPaletteSettings colorPaletteSettings(pConfig);
+    ColorPalette hotcuePalette = colorPaletteSettings.getHotcueColorPalette();
+    mixxx::RgbColor firstColor = mixxx::PredefinedColorPalettes::kDefaultCueColor;
+    DEBUG_ASSERT(hotcuePalette.size() > 0);
```

Comment:
How? That was my first attempt, but this would also crash:
```c++
    mixxx::RgbColor firstColor = hotcuePalette.at(0);
    VERIFY_OR_DEBUG_ASSERT(hotcuePalette.size() > 0) {
        firstColor = mixxx::PredefinedColorPalettes::kDefaultCueColor;
    }
```

And leaving `firstColor` uninitialized and adding an `else` case to `VERIFY_OR_DEBUG_ASSERT` is also not possible because `RgbColor` does not have a default constructor.